File: Java/Project/src/Math.java
package clam;
import java.util.Random;
import java.util.ArrayList;
import java.util.List;

public class Math {
	public static String mathing(String msg, List<String> keys) {
		Integer code = Integer.parseInt(msg);
		Integer A = Integer.parseInt(keys.get(0));
		Integer B = Integer.parseInt(keys.get(1));
		Integer msg_encrypt = (code * B) + A;
		return Integer.toString(msg_encrypt);
	}
	
	public static String unmath(String msg, List<Integer> keys) {
		Integer msg_encrypt = Integer.parseInt(msg);
		Integer A = keys.get(0);
		Integer B = keys.get(1);
		Integer msg_desencrypt = (msg_encrypt - A) / B;
		return Integer.toString(msg_desencrypt);
	}
	
	public static List<String> keys() {
		Random random = new Random();
		
	    List<String> numeros = new ArrayList<>();

	    numeros.add(Integer.toString(random.nextInt(1, 100)));
	    numeros.add(Integer.toString(random.nextInt(1, 100)));

	    return numeros;
	    }
}

File: Python/src/Math.py
import random

def mathing(msg: str, key_a: str, key_b: str) -> str:
    code = int(msg)
    a = int(key_a)
    b = int(key_b)
    msg_encrypt = (code * b) + a
    return str(msg_encrypt)


def unmathing(msg: str, key_a: str, key_b: str) -> str:
    code = int(msg)
    a = int(key_a)
    b = int(key_b)
    msg_dencrypt = (code - a) / b
    return str(int(msg_dencrypt))


def keys_generate() -> list[str]:
    key_a: int = random.randint(1, 100)
    key_b: int = random.randint(1, 100)
    return [str(key_a), str(key_b)]